src/Profiling.cpp

```diff
@@ -261,9 +261,17 @@ class InjectProfiling : public IRMutator {
         auto [new_extents, changed] = mutate_with_changes(op->extents);
         Expr condition = mutate(op->condition);
 
-        bool on_stack;
-        Expr size = compute_allocation_size(new_extents, condition, op->type, op->name, on_stack);
+        bool can_fit_on_stack;
+        Expr size = compute_allocation_size(new_extents, condition, op->type, op->name, can_fit_on_stack);
         internal_assert(size.type() == UInt(64));
+
+        bool on_stack = can_fit_on_stack;
```

Comment:
Is this just: 

`bool on_stack = can_fit_on_stack && !op->new_expr.defined();`

Reply:
Euhm, haha, yes... 😬 Will change.